commandapi-platforms/commandapi-bukkit/commandapi-bukkit-vh/src/main/java/dev/jorel/commandapi/CommandAPIVersionHandler.java

```diff
@@ -68,9 +70,54 @@ public interface CommandAPIVersionHandler {
 				case "1.20.3", "1.20.4" -> new NMS_1_20_R3();
 				case "1.20.5", "1.20.6" -> new NMS_1_20_R4();
 				case "1.21", "1.21.1" -> new NMS_1_21_R1();
-				default -> throw new UnsupportedVersionException(version);
+				default -> null;
 			};
+			if (platform != null) {
+				return platform;
+			} else {
+				if (CommandAPI.getConfiguration().shouldBeMoreLenientForMinorVersions()) {
+					return minecraftVersion.matchesPatch((NMS<?>) latestNMS);
+				}
+				throw new UnsupportedVersionException(version);
+			}
 		}
 	}
 
+	class Version {
+
+		private final String version;
+
+		// The minor version is something like 1.x
+		private final int minor;
```

Comment:
Shouldn't this be named major?

Reply:
Yeah, probably should revise all the names in the PR again. Threw it together fairly quickly and first had the Minecraft version treated as major.minor.patch and just deleted the major and patch fields.